Add macOS foreign configuration

plandes opened the pull request:
This adds macOS as a foreign configuration. I split it out as a separate package because:

* Provides for a cleaner separation from the `config` package as it is a significant chunk of code.
* Allows better a finer level of potential customization to clients.

This pull request also make the compilation command itself a global customizable variable (`*python-compile-command*`).  Doing so made the customization easier to write and allows clients to make other fine tuned adjustments that might be necessary for their respective platforms.

plandes committed:
make foreign library compile command configurable

plandes committed:
whitespace

plandes committed:
add macOS specific configuration

digikar99 commented on src/python-process.lisp:
```diff
@@ -177,7 +177,7 @@ can lead to memory leak.")))
       (float-features:with-float-traps-masked (:overflow :invalid)
         (ignore-some-conditions (floating-point-overflow floating-point-invalid-operation)
-          (import-module "numpy")
+	  (import-module "numpy")
```
Was there a bug here?

plandes commented:
@digikar99 Sorry, I thought I had put that back--but to answer your question: I am working on this part now.  When I run fiveam unit tests I've written for my own library from Roswell, this fails because `numpy` is not in Python's `sys.path`.  To make matter worse, `traceback` is not imported by the time the `numpy` library is imported so you get a failure without error reporting.

digikar99 commented:
I see, in that case, feel free to raise an issue!
The rest of the things here look good. Thank you for the PR!

digikar99 committed:
Remove (import-module "numpy") line in pystart

This wouldn't be a part of this PR. Hoping to fix any potential issue in separate commits!

plandes commented:
@digikar99 Thank you.  This has been resolved in the last pull request.  Let me know if you disagree with any of these changes as the evaluation is fairly different.  While I was not able to run your unit tests, I have my own that have passed.
